### PR TITLE
feat: add baidu map plugin

### PR DESCRIPTION
**Pull request by CoderSerio**
### Describe what this PR does / why we need it

Baidu Map is also one of the commonly used apps in daily life, so it is necessary to provide some support for it.

### Does this pull request fix one issue?

#98

### Describe how you did it

With Baidu Map Open APIs

### Describe how to verify it

Originally, there was an example here, but I noticed that this part of the file has been moved to [another repository](https://github.com/springaialibaba/spring-ai-alibaba-examples). I will also add it to the new one as soon as possible

image

The part of original example:
```java
@GetMapping("/baiduMap-get-address-weather")
public String baiDuMapGetAddressWeatherFunction(String input) {
    return chatClient.prompt()
        .functions("baiDuMapGetAddressWeatherFunction")
        .system("如果用户输入的内容中想询问天气情况,而且还给定了地址,则使用工具获取天气情况,不然提示用户缺少信息")
        .user(input)
        .call()
        .content();
}
```

### CODE REVIEW

**Comment by PolarishT** [comment]
Hi, this issue should related to #98

**Comment by CoderSerio** [comment]
> Hi, this issue should related to #98 

Thx noticing me, and I've added it😘

**Comment by CoderSerio** [comment]
@PolarishT 
I've implemented the feature to get detailed information about facilities around a specific address. 
And I've retained the ability to retrieve weather information, so both features can be used together.

image

Additionally, I've moved the dirs and files into `function-calling` from the original `plugins`

File: community/function-calling/spring-ai-alibaba-starter-function-calling-baidumap/pom.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            <artifactId>spring-ai-spring-boot-autoconfigure</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-configuration-processor</artifactId>
+            <optional>true</optional>
+        </dependency>
+
+        <dependency>
```
Comment (yuluo-yx) [inline community/function-calling/spring-ai-alibaba-starter-function-calling-baidumap/pom.xml:49]: 用一个序列化框架就可以了，fastjson 可以去掉 都用 jackson

Comment (CoderSerio) [reply community/function-calling/spring-ai-alibaba-starter-function-calling-baidumap/pom.xml:49]: 好的，今天晚上的时候我回去把这一段改掉

Comment (PolarishT) [reply community/function-calling/spring-ai-alibaba-starter-function-calling-baidumap/pom.xml:49]: 都是用jackson吗？那估计有些functionCalling Impl 得从 gson 转移到 jackjson了

File: community/function-calling/spring-ai-alibaba-starter-function-calling-baidumap/src/main/java/com/alibaba/cloud/ai/functioncalling/baidumap/BaiDuMapConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.context.annotation.Description;
+
+/**
+ * @author Carbon
+ */
+@Configuration
+@ConditionalOnClass(MapSearchService.class)
+@EnableConfigurationProperties(BaiDuMapProperties.class)
+@ConditionalOnProperty(prefix = "spring.ai.alibaba.functioncalling.baidumap", name = "enabled", havingValue = "true")
+public class BaiDuMapConfiguration {
```
Comment (PolarishT) [inline community/function-calling/spring-ai-alibaba-starter-function-calling-baidumap/src/main/java/com/alibaba/cloud/ai/functioncalling/baidumap/BaiDuMapConfiguration.java:36]: 命名不规范？这里按照规范应该是BaiDuMapAutoConfiguration

**Review by PolarishT — CHANGES_REQUESTED** [review]

**Comment by yuluo-yx** [comment]
看插件怎么定义了，我看用了 fastjson 所以顺带说了一下

**Comment by PolarishT** [comment]
统一使用JackSon吧～ 我们和Spring的生态对齐

**Comment by CoderSerio** [comment]
> 统一使用JackSon吧～ 我们和Spring的生态对齐

> 看插件怎么定义了，我看用了 fastjson 所以顺带说了一下

稍后我去改一下，辛苦二位的CR!

**Comment by CoderSerio** [comment]
Done. 辛苦大佬们有空再帮忙 CR 看下有没有什么问题。

**Review by PolarishT — APPROVED** [review]
LGTM

**Pull request merged**